### PR TITLE
Modify ftp_links script to insert galaxy_ids

### PR DESCRIPTION
**Pull request by simonbray**
@wm75 I couldn't really test this yet but this was roughly my idea

### CODE REVIEW

File: bioblend-scripts/ftp_links_to_yaml.py [hunk 1]
```diff
@@ -101,6 +123,15 @@ def records_to_yaml(records, collection_name):
         '-o', '--output',
         help='Write output to this file instead of to standard output'
     )
+    parser.add_argument(
+        '-i', '--history_id',
```
Comment (simonbray) [inline bioblend-scripts/ftp_links_to_yaml.py:127]: @wm75 it's probably easiest to use the same history all the time here, a kind of 'staging area' for dataset uploads

Comment (wm75) [reply bioblend-scripts/ftp_links_to_yaml.py:138]: Yes, I've added an ID for such a staging history to the job yml template.
Maybe there's a better way, but should be good enough for now.

File: bioblend-scripts/ftp_links_to_yaml.py [hunk 2]
```diff
@@ -14,40 +14,31 @@
...
+    while True:
         for link in ena_links:
             if ena_links_states[link] in NON_OK_TERMINAL_STATES:
-                ena_links_dataset_ids[link] = gi.tools.put_url(content=f"ftp://{link}", history_id=history_id)['outputs'][0]['id']
+                if ena_links_attempts_left[link] > 0:
+                    ena_links_dataset_ids[link] = gi.tools.put_url(
+                        content=f"ftp://{link}", history_id=history_id
+                    )['outputs'][0]['id']
+                    ena_links_attempts_left[link] -= 1
+            else:
```
Comment (simonbray) [inline bioblend-scripts/ftp_links_to_yaml.py:26]: needs indentation?

Comment (wm75) [reply bioblend-scripts/ftp_links_to_yaml.py:26]: uha, yes.

**Pull request merged**